<launch>
    <!-- Detection configuration -->
    <arg name="weights"               default="/home/heven/catkin_build_ws/src/Co_Deep_Project/best.pt"/>
    <arg name="data"                  default="/home/heven/catkin_build_ws/src/Co_Deep_Project/src/yolov5/data/coco128.yaml"/>
    <arg name="confidence_threshold"  default="0.40"/>
    <arg name="iou_threshold"         default="0.30"/>
    <arg name="maximum_detections"    default="1000"/>
    <arg name="device"                default="0"/>
    <arg name="agnostic_nms"          default="true"/>
    <arg name="line_thickness"        default="3"/>
    <arg name="dnn"                   default="true"/>
    <arg name="half"                  default="false"/>
    
    <!-- replace imgsz -->
    <arg name="inference_size_h"      default="960"/>
    <arg name="inference_size_w"      default="1280"/>
    
    <!-- Visualize using OpenCV window -->
    <arg name="view_image"            default="true"/>

    <!-- ROS topics -->
    <arg name="input_image_topic"       default="/carmaker_vds_client_node/image_raw/compressed"/>
    <arg name="output_topic"            default="/yolov5/detections"/>

    <!-- Optional topic (publishing annotated image) -->
    <arg name="publish_image"           default="true"/>
    <arg name="output_image_topic"      default="/yolov5/image_out"/>


    <node pkg="yolov5_ros" name="detector" type="detector.py" output="screen">
        <param name="weights"               value="$(arg weights)"/>
        <param name="data"                  value="$(arg data)"/>
        <param name="confidence_threshold"  value="$(arg confidence_threshold)"/>
        <param name="iou_threshold"         value="$(arg iou_threshold)" />
        <param name="maximum_detections"    value="$(arg maximum_detections)"/>
        <param name="device"                value="$(arg device)" />
        <param name="agnostic_nms"          value="$(arg agnostic_nms)" />
        <param name="line_thickness"        value="$(arg line_thickness)"/>
        <param name="dnn"                   value="$(arg dnn)"/>
        <param name="half"                  value="$(arg half)"/>

        <param name="inference_size_h"      value="$(arg inference_size_h)"/>
        <param name="inference_size_w"      value="$(arg inference_size_w)"/>

        <param name="input_image_topic"     value="$(arg input_image_topic)"/>
        <param name="output_topic"          value="$(arg output_topic)"/>

        <param name="view_image"            value="$(arg view_image)"/>

        <param name="publish_image"         value="$(arg publish_image)"/>
        <param name="output_image_topic"    value="$(arg output_image_topic)"/>
    </node>

    <node pkg="yolov5_ros" name="fusion_node" type="matrix_fusion.py" output="screen"/>

    <!-- <include file="$(find camera_launch)/launch/d435.launch"/> -->


</launch>
